vendor/build_path_prefix_map/src/build_path_prefix_map.mli

```diff
@@ -15,10 +41,29 @@ type map = pair option list
 val encode_map : map -> string
 val decode_map : string -> (map, error_message) result
 
-val rewrite_opt : map -> path -> path option
-(** [rewrite_opt map path] tries to find a source in [map]
+val rewrite_first : map -> path -> path option
+(** [rewrite_first map path] tries to find a source in [map]
     that is a prefix of the input [path]. If it succeeds,
     it replaces this prefix with the corresponding target.
     If it fails, it just returns [None]. *)
 
+val rewrite_all : map -> path -> path list
+(** [rewrite_all map path] finds all sources in [map]
```

Comment:
Are you pulling these changes from upstream? If so, this should be done separately.

Reply:
As I mentioned, I have a PR to @gasche, to get this change into his upstream repo. I've not gotten any response from him yet on that PR, however, he was the one that merged that change into the OCaml trunk. I suppose if you wanted to merge this into Dune separately you could, and then when it gets upstream you could go back to getting it there.

Reply:
I've not heard from @gasche on the build_path_prefix_map changes, so we may need to merge this as part of the PR and later vendor it when it is ready. Relative to my build_path_prefix_map I have one change, a local copy of List.find_map, in order to be able to build Dune with versions of OCaml that did not yet have List.find_map.